pyparsing 3.0.8 causing type errors for rdflib

When using pyparsing 3.0.8 we are getting the following type errors on our master branch.

```
$ .venv/bin/python3 -m mypy --show-error-context --show-error-codes
rdflib/plugins/sparql/parser.py:200: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:391: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:440: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:481: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:500: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:556: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:560: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:563: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:567: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:597: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:603: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:609: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:612: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:613: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:631: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:632: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:662: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:675: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:689: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:713: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:736: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1089: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1156: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1171: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1179: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1230: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1232: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1280: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1287: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1288: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1292: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1344: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1347: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1348: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1389: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1394: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1395: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1396: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1397: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1404: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1427: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1462: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1493: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1500: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/parser.py:1501: error: "_SpecialForm" not callable  [operator]
rdflib/plugins/sparql/results/tsvresults.py:44: error: "_SpecialForm" not callable  [operator]
Found 46 errors in 2 files (checked 284 source files)
```

Files in question are:

- https://github.com/RDFLib/rdflib/blob/1cba9d87c84b40252fcd7bed9d566839c77e078a/rdflib/plugins/sparql/parser.py
- https://github.com/RDFLib/rdflib/blob/1cba9d87c84b40252fcd7bed9d566839c77e078a/rdflib/plugins/sparql/results/tsvresults.py

Errors can be seen in: https://github.com/RDFLib/rdflib/runs/5962843175?check_suite_focus=true

I think the problem somehow relates to `pyparsing.Optional` being confused with `typing.Optional`, but I'm not sure about this, I tried to alias it to something else and tried to use it with package name and it did not help. Not sure why this is happening, but I will try figure out. Just recording this while I work on it.

--- a/pyparsing/exceptions.py
+++ b/pyparsing/exceptions.py
@@ -2,7 +2,7 @@
 
 import re
 import sys
-from typing import Optional
+import typing
 
 from .util import col, line, lineno, _collapse_string_to_ranges
 from .unicode import pyparsing_unicode as ppu
@@ -25,7 +25,7 @@ class ParseBaseException(Exception):
         self,
         pstr: str,
         loc: int = 0,
-        msg: Optional[str] = None,
+        msg: typing.Optional[str] = None,
         elem=None,
     ):
         self.loc = loc

